sdk/src/main/java/com/mapbox/maps/MapProvider.kt

```diff
@@ -50,7 +52,10 @@ internal object MapProvider {
         paramsProvider(context, accessToken, MapboxModuleType.MapTelemetry)
       }
     }
-    mapTelemetry.onAppUserTurnstileEvent()
+    // Report turnstile event in background thread.
+    thread {
+      mapTelemetry.onAppUserTurnstileEvent()
```

Comment:
just curious. is it possible, that `mapTelemetry` might be not initialized before this thread is started? Maybe thread execute `onAppUserTurnstileEvent` before line 51

Reply:
`MapboxModuleProvider.createModule` is synchronous so `mapTelemetry` should be valid when we hit the `mapTelemetry.onAppUserTurnstileEvent() `call.

Reply:
but what if compiler decide to change order of calls and block `thread { }` will runs first?

Reply:
and from documentation `thread { }` block has a property `start` 

`if true, the thread is immediately started` 

so it might be a little chance that thread will start with non initialized function

Reply:
> but what if compiler decide to change order of calls and block thread { } will runs first?

I assume newly started thread is guaranteed to see the latest values of all the variables that were set by the thread that have started it, e.g. from [docs](https://docs.oracle.com/javase/6/docs/api/java/util/concurrent/package-summary.html#MemoryVisibility) - `A call to start on a thread happens-before any action in the started thread.`

Reply:
Have checked with http interceptor, and event was successfully sent in both foreground and background threads.
Not sure @alexander-kulikovskii why the compiler would change the order of calls? Do you mean the thread executes before `mapTelemetry` is created? Can you give a code sample?

Reply:
@paulrenn67 in general compiler might apply various optimizations and reorder instructions (e.g. [docs](https://docs.oracle.com/javase/specs/jls/se8/html/jls-17.html#jls-17.4-A)), but this should no be a problem here since [docs](https://docs.oracle.com/javase/6/docs/api/java/util/concurrent/package-summary.html#MemoryVisibility) also say that `A call to start on a thread happens-before any action in the started thread.`